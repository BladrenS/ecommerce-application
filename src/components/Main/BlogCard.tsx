import type { FC } from 'react';
import { NavLink } from 'react-router-dom';

import type { BlogItem } from '../../constants';
import styles from './styles.module.scss';

type BlogCardProps = BlogItem & {
  index: number;
};

export const BlogCard: FC<BlogCardProps> = ({ date, title, text, image, index, link }) => {
  const content = (
    <>
      <img className={styles['blog-img']} src={image} alt={title} />
      <div className={styles['blog-info']}>
        <span className={styles['info-date']}>{date}</span>
        <span className={styles['info-header']}>{title}</span>
        <span className={styles['info-text']}>{text}</span>
      </div>
    </>
  );

  return (
    <NavLink to={link ? `${link}` : `/article/${index}`} className={styles['blog-item']}>
      {content}
    </NavLink>
  );
};
