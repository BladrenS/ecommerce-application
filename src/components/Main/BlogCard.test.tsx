import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { BlogCard } from './BlogCard';

const mockBlogItem = {
  date: '01.06.2025',
  title: 'Новый пост в блоге',
  text: 'Это краткое описание блога.',
  image: '/test-image.jpg',
};

describe('BlogCard snapshot test', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <BlogCard {...mockBlogItem} index={3} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

test('renders blog card content', () => {
  render(<BlogCard {...mockBlogItem} index={5} />, { wrapper: MemoryRouter });

  expect(screen.getByAltText(mockBlogItem.title)).toBeInTheDocument();
  expect(screen.getByText(mockBlogItem.date)).toBeInTheDocument();
  expect(screen.getByText(mockBlogItem.title)).toBeInTheDocument();
  expect(screen.getByText(mockBlogItem.text)).toBeInTheDocument();
});

test('blog card links to correct article', () => {
  render(<BlogCard {...mockBlogItem} index={7} />, { wrapper: MemoryRouter });

  const link = screen.getByRole('link');
  expect(link).toHaveAttribute('href', '/article/7');
});
