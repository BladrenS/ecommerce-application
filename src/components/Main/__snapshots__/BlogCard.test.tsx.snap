// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard snapshot test renders correctly 1`] = `
<DocumentFragment>
  <a
    class="blog-item"
    data-discover="true"
    href="/article/3"
  >
    <img
      alt="Новый пост в блоге"
      class="blog-img"
      src="/test-image.jpg"
    />
    <div
      class="blog-info"
    >
      <span
        class="info-date"
      >
        01.06.2025
      </span>
      <span
        class="info-header"
      >
        Новый пост в блоге
      </span>
      <span
        class="info-text"
      >
        Это краткое описание блога.
      </span>
    </div>
  </a>
</DocumentFragment>
`;
